test/hs/Test/Ganeti/TestCommon.hs

```diff
@@ -118,15 +118,20 @@ import Numeric
 
 import qualified Ganeti.BasicTypes as BasicTypes
 import Ganeti.JSON (ArrayObject(..))
+import Ganeti.Objects (TagSet)
 import Ganeti.Types
 import Ganeti.Utils.Monad (unfoldrM)
 
 -- * Arbitrary orphan instances
 
+#if !MIN_VERSION_QuickCheck(2,8,0)
 instance (Ord k, Arbitrary k, Arbitrary a) => Arbitrary (M.Map k a) where
```

Comment:
Adding comment here but it's a comment on this entire commit. This works as a stop-gap, but it's the wrong fix. Let's not delay this patch series because reviewing N commits is tiring, I'll send in the weekend a proper fix with explanation.